src/rgw/rgw_data_sync.cc

```diff
@@ -1097,10 +1116,14 @@ class RGWDataSyncSingleEntryCR : public RGWCoroutine {
         }
       }
       /* FIXME: what do do in case of error */
-      if (marker_tracker && !entry_marker.empty()) {
+      if (sync_status == 0 && !marker_handle->marker.empty()) {
```

Comment:
if errors happened，sync_status will < 0，and the entry marker will leave in finish_markers,
which lead the finish_markers hold lots of entries. And also the sync marker can't be updated untill radosgw restarted.

Reply:
good catch, i'll look into that

Reply:
@cbodley Hi casey， do you have any fix idea for this issue?

Reply:
@cbodley